### ISSUE TITLE
Ng-selectize removes initial set of ngModel

### ISSUE THREAD

**Issue by stefannikolic** [NicholasAzar/ng-selectize#52]
Hi,

I have a problem with `<ng-selectize>` removes my initially set ngModel.
I have a code like this

`HTML`
`        <ng-selectize [config]="config" [options]="myOptions" placeholder="Select" [(ngModel)]="myObject.type"></ng-selectize>
`

and in component I have:

`TS`
`this.myObject = {
    type: 'myValue1'
};`
`this.myOptions = [{
			value: 'myValue1',
			label: 'myLabel1'
		}, {
			value: 'myValue2',
			label: 'myLabel2'
		}];`
`config: any = {
		labelField: 'label',
		valueField: 'value',
		maxItems: 1
	};`

So what happens is that my dropdown renders myLabel1 correctly (so it picked it up from an object) but afterwards it removes it and I got

`this.myObject = {
    type: ''
};`

The dropdown still renders myLabel1 but object doesn't contain property value anymore.

When I am reassigning myObject.type (choosing from dropdown) it is ok but initially it is removed.

This is only happening when I am rendering <ng-selectize> tag in HTML, if I remove the tag myObject.type is correct. Also it is correct if I set in config that maxItems are more than 1.

So do you maybe know what can be an issue here, that I am getting initially removed ngModel (that is tied to ng-selectize) from myObject upon initializing the page?

All the best.

Stefan

**Comment by khan8698** [comment]
Hi @stefannikolic !
Did you get any solution for this problem i am facing the same problem right now and i am still clueless about it. by the way interesting part is that if i load data after some delay (from API call) then it is working perfectly but creates this problem when it is loaded immediately.
One of the trick i try to use that i assign selected value (ngModel value) after assigning the data but that doesn't seemed to be working.
Please let me know that if there is any solution for that.
Thanks in advance.

Regards,
Khan

**Comment by stefannikolic** [comment]
Hi @khan8698 I used all the "solutions" you mentioned and they worked but I didn't like them. I think that problem is somehow in that I am giving to selectize array of objects and tell it that it's selection should be stored as a string (you can see my code), so I think that library is not doing the conversion right. I suppose it should take the id of selected option and put it in the property you specified as ngModel.

**Comment by khan8698** [comment]
In My case the value and label both are string but this problem is still happening. i also didn't like much this approach (assigning the model value again after some time). but nothing seemed to be working except this

**Comment by stefannikolic** [comment]
In my case also the label and the values are strings. But I think that the problem is that you are giving array of objects to the selectize (in my case this.myOptions) and ngModel is string, so maybe if this.myOptions is an array of strings it will work or ngModel (in my case this.myObject.type) is an object.

**Comment by khan8698** [comment]
May it is expecting an array because it is working perfectly in case maxItem more than 1.

**Comment by khan8698** [comment]
@NicholasAzar any update related to this issue ?

**Comment by NicholasAzar** [comment]
Hi @stefannikolic and @khan8698 , could you please confirm if this is still an issue in v1.1.0? I'm trying out your example and seems to be working as expected.. let me know.

html:
```
<ng-selectize [config]="config" [options]="myOptions" placeholder="Select" [(ngModel)]="myObject.type"></ng-selectize>

<pre>Current value: {{myObject | json}}</pre>
```

ts:
```
this.myObject = { type: 'myValue1' };
this.myOptions = [{ value: 'myValue1', label: 'myLabel1' }, { value: 'myValue2', label: 'myLabel2' }];
this.config = { labelField: 'label', valueField: 'value', maxItems: 1 };
```

**Comment by khan8698** [comment]
@NicholasAzar I just updated library to version 1.1.0 and it started throwing an error in console leaving complete component broken. Here is the Error

`OrderFormComponent.html:4 ERROR Error: Cannot find a differ supporting object '[object Object]' of type 'object'`

While it was working fine before. Can you please check that it.
Thanks

**Comment by NicholasAzar** [comment]
@khan8698 The only way i could reproduce this was to have the [options] as an object rather then an array of objects.. is this the case?

**Comment by khan8698** [comment]
@NicholasAzar I don't think that this is the problem. code was working correctly in version 1.0.0 but started producing error in version 1.1.0.
Let me add sample options and configurations to make things clear. (this is working with version 1.0.0 obviously with the issue of emptying model field but not on version 1.1.0 ).

`dropDownOptions = [{
      "label": "label1",
      "value": "value1"
    }, {
      "label": "label2",
      "value": "value2"
    }]`

`commonSelectizeConfig= {
      labelField: 'label',
      searchField: ['label'],
      sortField: 'label',
      loadThrottle: 1000,
      dropdownDirection: 'down',
      plugins: ['dropdown_direction', 'remove_button'],
      onBlur: function () {},   //some function      
      onLoad: function () {}  //some function      
      }`

HTML:

`<ng-selectize (ngModelChange)="getData($event); getSecondDropDownData($event) "
                    [(ngModel)]="params.account" name="account" [config]="commonSelectizeConfig"
                    [options]="dropDownOptions "></ng-selectize>`

Interestingly it is working for some drop down (those who require data from API) while not for some (those who have local / hard coded  data).

**Comment by stefannikolic** [comment]
@NicholasAzar sorry I found some other solution so didn't get back to this one. I don't think that there is a bug but the feature is not like some others drop-down libraries. When mapping is 1 to 1 (as I explained above) works fine, but unfortunately, it didn't work out for me that way so I had to change it.

**Comment by khan8698** [comment]
@stefannikolic what was the work around solution ? can you please tell me may b i am able to make that work until issue would be resolved fully on my side too ?

**Comment by paesku** [comment]
After updating from **v1.0.11** to **v1.0.11** we faced the problem, that our model presets weren't rendered properly. Which means, they popped up and disappeared after seconds and then the user couldn't select anything. Data are there and we didn't really figure out what causes this problems. We went back to **v1.0.11**.
